Elbow operator (#7472)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

* Two new functions called :func:`~.math.convert_to_su2` and :func:`~.math.convert_to_su4` have been added to `qml.math`, which convert unitary matrices to SU(2) or SU(4), respectively, and optionally a global phase.
  [(#7211)](https://github.com/PennyLaneAI/pennylane/pull/7211)

> * A new template :class:`~.TemporaryAND` has been added. The ``TemporaryAND`` operator is a three-qubit gate equivalent to an AND, or reversible Toffoli, gate that leverages extra information
>   about the target wire to enable a more efficient decomposition. It is useful for an efficient decomposition of the :class:`~.Select` template, for example. 
>   [(#7472)](https://github.com/PennyLaneAI/pennylane/pull/7472)

Reviewer: ```suggestion
* A new template :class:`~.TemporaryAND` has been added. The :class:`~.TemporaryAND` (a.k.a.  :class:`~.Elbow`)
  operator is a three-qubit gate which can serve as an equivalent to an AND, or reversible Toffoli, gate.
  It leverages extra information about the target wire to enable more efficient decompositions.
  :class:`~.TemporaryAND` is useful for an efficient decomposition of the :class:`~.Select` template, for example. 
  [(#7472)](https://github.com/PennyLaneAI/pennylane/pull/7472)
```

* Two new functions called :func:`~.math.convert_to_su2` and :func:`~.math.convert_to_su4` have been added to `qml.math`, which convert unitary matrices to SU(2) or SU(4), respectively, and optionally a global phase.
  [(#7211)](https://github.com/PennyLaneAI/pennylane/pull/7211)

* A new template :class:`~.TemporaryAND` has been added. The :class:`~.TemporaryAND` (a.k.a.  :class:`~.Elbow`)
  operator is a three-qubit gate which can serve as an equivalent to an AND, or reversible Toffoli, gate.
  It leverages extra information about the target wire to enable more efficient decompositions.
  :class:`~.TemporaryAND` is useful for an efficient decomposition of the :class:`~.Select` template, for example. 
  [(#7472)](https://github.com/PennyLaneAI/pennylane/pull/7472)

  ```python
  dev = qml.device("default.qubit", shots=1)
  @qml.qnode(dev)
  def circuit():
      qml.X(0)
      qml.X(1)
      qml.TemporaryAND([0,1,2])
      qml.CNOT([2,3])
      # We can now apply the adjoint TemporaryAND,

>       # because after applying a Toffoli, the target wire would be |0>.

Author: I could just add the states during the process for simplicity, would that work?

```
import pennylane as qml

dev = qml.device("default.qubit", shots=1)
@qml.qnode(dev)
def circuit():
    # |0000⟩
    qml.X(0) # |1000⟩
    qml.X(1) # |1100⟩
    qml.TemporaryAND([0,1,2], control_values = (10,1)) # |1110⟩
    qml.CNOT([2,3]) # |1111⟩
    qml.adjoint(qml.TemporaryAND([0,1,2])) # |1101⟩
    return qml.sample(wires=[0,1,2,3])

print(circuit())
```

* Two new functions called :func:`~.math.convert_to_su2` and :func:`~.math.convert_to_su4` have been added to `qml.math`, which convert unitary matrices to SU(2) or SU(4), respectively, and optionally a global phase.
  [(#7211)](https://github.com/PennyLaneAI/pennylane/pull/7211)

* A new template :class:`~.TemporaryAND` has been added. The :class:`~.TemporaryAND` (a.k.a.  :class:`~.Elbow`)
  operator is a three-qubit gate which can serve as an equivalent to an AND, or reversible Toffoli, gate.
  It leverages extra information about the target wire to enable more efficient decompositions.
  :class:`~.TemporaryAND` is useful for an efficient decomposition of the :class:`~.Select` template, for example. 
  [(#7472)](https://github.com/PennyLaneAI/pennylane/pull/7472)

  ```python
  dev = qml.device("default.qubit", shots=1)
  @qml.qnode(dev)
  def circuit():
      # |0000⟩
      qml.X(0) # |1000⟩
      qml.X(1) # |1100⟩
      # The target wire is in state |0>, so we can apply TemporaryAND
      qml.TemporaryAND([0,1,2]) # |1110⟩
      qml.CNOT([2,3]) # |1111⟩

>       # Target wire will be in state |0> after AND gate, so we can apply adjoint(TemporaryAND)

Reviewer 2: ```suggestion
      # The target wire will be in state |0> after adjoint(TemporaryAND) gate is applied, so we can apply adjoint(TemporaryAND)
```

* Two new functions called :func:`~.math.convert_to_su2` and :func:`~.math.convert_to_su4` have been added to `qml.math`, which convert unitary matrices to SU(2) or SU(4), respectively, and optionally a global phase.
  [(#7211)](https://github.com/PennyLaneAI/pennylane/pull/7211)

* A new template :class:`~.TemporaryAND` has been added. The ``TemporaryAND`` operation is a three-qubit gate equivalent to an ``AND``, or reversible :class:`~pennylane.Toffoli`, gate that leverages extra information
  about the target wire to enable more efficient circuit decompositions. The ``TemporaryAND`` assumes the target qubit
  to be initialized in ``|0〉``, while the ``Adjoint(TemporaryAND)`` assumes the target output to be ``|0〉``.
  For more details, see Fig. 4 in `arXiv:1805.03662 <https://arxiv.org/abs/1805.03662>`_.
  :class:`~.TemporaryAND` is useful for an efficient decomposition of the :class:`~.Select` template, for example. 

>   [(#7472)](https://github.com/PennyLaneAI/pennylane/pull/7472)

Reviewer: Let's make sure to mention the alias here as well.
```suggestion
* A new template :class:`~.TemporaryAND` has been added. The  :class:`~.TemporaryAND` (a.k.a.  :class:`~.Elbow`)
  operation is a three-qubit gate equivalent to an ``AND``, or reversible :class:`~pennylane.Toffoli`, gate
  that leverages extra information about the target wire to enable more efficient circuit decompositions.
  The ``TemporaryAND`` assumes the target qubit to be initialized in ``|0〉``, while the ``Adjoint(TemporaryAND)`` assumes the target output to be ``|0〉``.
  For more details, see Fig. 4 in `arXiv:1805.03662 <https://arxiv.org/abs/1805.03662>`_.
  :class:`~.TemporaryAND` is useful for an efficient decomposition of the :class:`~.Select` template, for example. 
  [(#7472)](https://github.com/PennyLaneAI/pennylane/pull/7472)
```
